bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -151,23 +146,26 @@ where
 			}
 		}
 
-		let topic = blake2_256(&(INBOUND_QUEUE_TOPIC_PREFIX, message.nonce).encode());
+		// Add SetTopic instruction if not already present as the last instruction
+		if !matches!(remote_xcm.0.last(), Some(SetTopic(_))) {
+			remote_xcm.0.push(SetTopic(unique(&remote_xcm)));
+		}
+
```

Comment:
As we discussed, is that supposed to revert to
```
let topic = blake2_256(&(INBOUND_QUEUE_TOPIC_PREFIX, message.nonce).encode());
```
which provides determinism.

Reply:
agreed

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/32/commits/c1484452b34459d6fbad7dfc4577a297397d03e6 and https://github.com/vgeddes/polkadot-sdk/pull/32/commits/db80fd5148ff1adc99a81821b3438b01af2070cc